homeassistant/components/wiffi/config_flow.py

```diff
@@ -50,7 +54,10 @@ async def async_step_user(self, user_input=None):
     @callback
     def _async_show_form(self, errors=None):
         """Show the config flow form to the user."""
-        data_schema = {vol.Required(CONF_PORT, default=DEFAULT_PORT): int}
+        data_schema = {
+            vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+            vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): int,
```

Comment:
Optional options should not be part of the main flow but an options flow.

Is it impossible to detect the correct update interval that should be set from the device api?

Reply:
Thanks, that's a good point. The timeout is better off in a option flow than in the main config flow.

Unfortunately there is no easy way to get the configured update interval. The only way I know is to scrape it from the HTML site, which is cumbersome. The other issue is that the devices do not always respect the configured update interval and therefore a more relaxed timeout setting is required.